js/indicators/ichimoku-kinko-hyo.src.js

```diff
@@ -396,50 +485,203 @@ seriesType('ikh', 'sma',
+                                points[0] = points[0].concat(sectionPoints);
+                                nextPoints[0] =
+                                    nextPoints[0].concat(sectionNextPoints);
+                            } else {
+                                points[1] = points[1].concat(sectionPoints);
+                                nextPoints[1] =
+                                    nextPoints[1].concat(sectionNextPoints);
+                            }
+                        }
+                    } else {
+                        // compare first point of the section
+                        if (sectionPoints[0].plotY >
+                            sectionNextPoints[0].plotY
+                        ) {
+                            points[0] = points[0].concat(sectionPoints);
+                            nextPoints[0] =
+                                nextPoints[0].concat(sectionNextPoints);
+                        } else {
+                            points[1] = points[1].concat(sectionPoints);
+                            nextPoints[1] =
+                                nextPoints[1].concat(sectionNextPoints);
+                        }
+                    }
+                }
+
+                // render color and negativeColor paths
+                each(['graphsenkouSpanColor', 'graphsenkouSpanNegativeColor'],
+                    function (areaName, i) {
+                        if (points[i].length && nextPoints[i].length) {
+                            indicator.points = points[i];
+                            indicator.nextPoints = nextPoints[i];
+                            indicator.color = (i === 0) ?
+                                color :
+                                negativeColor;
+
+                            indicator.options = H.merge(
+                                mainLineOptions.senkouSpan.styles,
+                                gappedExtend
+                            );
+                            indicator.graph = indicator[areaName];
+                            indicator.fillGraph = true;
+
+                            SMA.prototype.drawGraph.call(indicator);
+
+                            // Now save line
+                            indicator[areaName] = indicator.graph;
+                            indicator.graphCollection.push(areaName);
+                        }
+                    });
+
+            } else {
+                // when user set only senkouSpan style.fill property
+                indicator.points = ikhMap.senkouSpanB;
+                indicator.options = merge(
+                    mainLineOptions.senkouSpan.styles,
+                    gappedExtend
+                );
+                indicator.graph = indicator.graphsenkouSpan;
+                indicator.nextPoints = ikhMap.senkouSpanA;
+
+                indicator.fillGraph = true;
+                indicator.color = color;
+                SMA.prototype.drawGraph.call(indicator);
+
```

Comment:
Lines 669 - 679 are the same like 646 - 662 (with one extra color option. 
I would consider to extract it as external function and avoid duplicate of code.